### ISSUE TITLE
SDL_CreateWindow

### ISSUE THREAD

**Issue by lzavoianu** [emscripten-core/emscripten#8732]
Error is thrown when creating Window:

exception thrown: SyntaxError: '' is not a valid selector,__findEventTarget
__findCanvasEventTarget
_emscripten_set_canvas_element_size
_Emscripten_CreateWindow
_SDL_CreateWindow
_main
Module._main
callMain

Even with the code from the tests:

```
#include <stdio.h>
#include <SDL2/SDL.h>


int main() {
    SDL_Init(SDL_INIT_VIDEO);
    SDL_Window *window;
    SDL_Renderer *renderer;
    SDL_CreateWindowAndRenderer(256, 256, 0, &window, &renderer);
    
    SDL_Surface *screen = SDL_CreateRGBSurface(0, 256, 256, 8, 0, 0, 0, 0);
    
    if (SDL_MUSTLOCK(screen)) SDL_LockSurface(screen);
    if (SDL_MUSTLOCK(screen)) SDL_UnlockSurface(screen);
    SDL_RenderPresent(renderer);
    
    SDL_Quit();
    
    return 0;
}
```

**Comment by kripken** [comment]
This may be related to the recent change in the HTML5 API. The web console shows
```
Rules for selecting event targets in HTML5 API are changing: instead of using document.getElementById() that only can refer to elements by their DOM ID, new event target selection mechanism uses the more flexible function document.querySelector() that can look up element names, classes, and complex CSS selectors. Build with -s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=1 to change to the new lookup rules. See https://github.com/emscripten-core/emscripten/pull/7977 for more details. 
```
See #7977 for more details.

**Comment by gabrielcuvillier** [comment]
I can confirm **DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0** is fixing the issue, as I had the same problem recently. I guess SDL2 have to be fixed to support the new behavior

**Comment by kripken** [comment]
cc @Daft-Freak - I think we fixed that in https://github.com/emscripten-ports/SDL2/pull/75 ? Was there more work that needed to get done?

**Comment by Daft-Freak** [comment]
That should be enough, but we haven't updated the port to include that commit yet. It currently causes some test failures though (#8733). (Noticed while updating the version (#8572).)

**Comment by kripken** [comment]
@Daft-Freak thanks, ok, sounds like from the last comment there that @juj may have fixed this meanwhile, so hopefully we can do that update!

**Comment by Qbz23** [comment]
I'm also encountering this issue. If I understand correctly, the fix should just be adding 

-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0

to the flags in my cmake file?

**Comment by gabrielcuvillier** [comment]
@Qbz23, yes you need to add exactly this in your cmake file, at least at the link phase

**Comment by Qbz23** [comment]
ah yeah that worked for me thanks. onto the next issue figuring out linking/dlopen error

**Comment by Daft-Freak** [comment]
This should be fixed by #8572

**Comment by lzavoianu** [comment]
Confirmed. Fixed by #8572
